mender-auth/ipc/server.cpp

```diff
@@ -18,93 +18,59 @@
-				return;
-			}
-			auto req = exp_req.value();
+namespace auth_client = mender::api::auth;
+namespace error = mender::common::error;
+namespace expected = mender::common::expected;
+namespace dbus = mender::common::dbus;
 
-			auto exp_resp = req->MakeResponse();
-			if (!exp_resp) {
-				log::Error("Failed to create the response: " + exp_resp.error().message);
-				return;
-			}
-			auto resp = exp_resp.value();
-
-			log::Debug("Got path: " + req->GetPath());
+using namespace std;
 
-			if (req->GetPath() == "/getjwttoken") {
-				resp->SetHeader("X-MEN-JWTTOKEN", this->GetJWTToken());
-				resp->SetHeader("X-MEN-SERVERURL", this->GetServerURL());
-				resp->SetStatusCodeAndMessage(200, "Success");
-				resp->AsyncReply([](error::Error err) {
-					if (err != error::NoError) {
-						log::Error("Failed to reply: " + err.message);
-					};
-				});
-			} else if (req->GetPath() == "/fetchjwttoken") {
-				auto err = auth_client::FetchJWTToken(
-					this->client_,
-					this->server_url_,
-					private_key_path,
-					identity_script_path == "" ? default_identity_script_path_
-											   : identity_script_path,
-					[this](auth_client::APIResponse resp) {
-						this->CacheAPIResponse(this->server_url_, resp);
-					},
-					this->tenant_token_);
-				if (err != error::NoError) {
-					log::Error("Failed to set up the fetch request: " + err.message);
-					resp->SetStatusCodeAndMessage(500, "Internal Server Error");
-					resp->AsyncReply([](error::Error err) {
-						if (err != error::NoError) {
-							log::Error("Failed to reply: " + err.message);
-						};
-					});
-					return;
-				}
-				resp->SetStatusCodeAndMessage(200, "Success");
-				resp->AsyncReply([](error::Error err) {
-					if (err != error::NoError) {
-						log::Error("Failed to reply: " + err.message);
-					};
-				});
-			} else {
-				resp->SetStatusCodeAndMessage(404, "Not Found");
-				resp->AsyncReply([](error::Error err) {
-					if (err != error::NoError) {
-						log::Error("Failed to reply: " + err.message);
-					};
-				});
+// Register DBus object handling auth methods and signals
+error::Error Caching::Listen(const string &private_key_path, const string &identity_script_path) {
+	auto dbus_obj = make_shared<dbus::DBusObject>("/io/mender/AuthenticationManager");
+	dbus_obj->AddMethodHandler<dbus::ExpectedStringPair>(
+		"io.mender.AuthenticationManager", "io.mender.Authentication1", "GetJwtToken", [this]() {
+			return dbus::StringPair {GetJWTToken(), GetServerURL()};
+		});
+	dbus_obj->AddMethodHandler<expected::ExpectedBool>(
+		"io.mender.AuthenticationManager",
+		"io.mender.Authentication1",
+		"FetchJwtToken",
+		[this, private_key_path, identity_script_path]() {
+			auto err = auth_client::FetchJWTToken(
+				client_,
+				server_url_,
+				private_key_path,
+				identity_script_path == "" ? default_identity_script_path_ : identity_script_path,
+				[this](auth_client::APIResponse resp) {
+					CacheAPIResponse(server_url_, resp);
```

Comment:
Q: Should we log an error in case `!resp`?

Reply:
Not a bad idea :grin: